From 0d4aa51c433b239fbebacb86d8ad2cec46a1137e Mon Sep 17 00:00:00 2001
From: Han Wang <zjuwanghan@outlook.com>
Date: Sun, 15 Aug 2021 20:48:24 +0800
Subject: [PATCH -next] md: remove useless INIT_LIST_HEAD()

List candidates has been inited through LIST_HEAD, so remove
needless INIT_LIST_HEAD(&candidates) in autorun_devices.

Signed-off-by: Han Wang <zjuwanghan@outlook.com>
---
 drivers/md/md.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index ae8fe54ea..fc382bce5 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -6485,7 +6485,6 @@ static void autorun_devices(int part)
 					 struct md_rdev, same_set);
 
 		pr_debug("md: considering %s ...\n", bdevname(rdev0->bdev,b));
-		INIT_LIST_HEAD(&candidates);
 		rdev_for_each_list(rdev, tmp, &pending_raid_disks)
 			if (super_90_load(rdev, rdev0, 0) >= 0) {
 				pr_debug("md:  adding %s ...\n",
-- 
2.25.1

